src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerEngine.java

```diff
@@ -242,8 +288,16 @@ private void initialize(boolean createBamOutIndex, final boolean createBamOutMD5
 
         initializeActiveRegionEvaluationGenotyperEngine();
 
-        genotypingEngine = new HaplotypeCallerGenotypingEngine(hcArgs, samplesList, ! hcArgs.doNotRunPhysicalPhasing, hcArgs.applyBQD);
-        genotypingEngine.setAnnotationEngine(annotationEngine);
+        defaultGenotypingEngine = new HaplotypeCallerGenotypingEngine(hcArgs, samplesList, ! hcArgs.doNotRunPhysicalPhasing, hcArgs.applyBQD);
+        defaultGenotypingEngine.setAnnotationEngine(annotationEngine);
+
+        // Create other custom genotyping engines if user provided ploidyRegions
+        for (final int ploidy : this.allCustomPloidies) {
```

Comment:
so currently this code maintains two separate genotypers for ployidy 2 if it gets specified?